src/libraries/System.Data.Common/src/System/Data/Common/DbProviderFactories.cs

```diff
@@ -188,7 +193,7 @@ private static DbProviderFactory GetFactoryInstance(Type providerFactoryClass)
             return (DbProviderFactory)factory;
         }
 
-
+        [RequiresUnreferencedCode(DataSet.RequiresUnreferencedCodeMessage)]
```

Comment:
Same here, we should probably adjust message to just say here that the reason why this is unsafe is because we are calling Type.GetType(string)